core/network/network.go

```diff
@@ -194,6 +194,9 @@ type Dialer interface {
 	// Notify/StopNotify register and unregister a notifiee for signals
 	Notify(Notifiee)
 	StopNotify(Notifiee)
+
+	// CanDial returns whether the dialer can dial peer p at addr
```

Comment:
It means we _will_ dial when true and we _wont_ dial when false. 

In the case of autonatv2 swarm use a read only black hole detector(because there may be many false negatives) so whenever we are not sure about our connectivity we will not attempt to dial from the autonatv2 swarm.